sqlest/src/main/scala/sqlest/executor/Database.scala

```diff
@@ -118,29 +118,35 @@ class Session(database: Database) extends Logging {
     withConnection { connection =>
       val (preprocessedSelect, sql, argumentLists) = database.statementBuilder(select)
       val startTime = new DateTime
-      val preparedStatement = prepareStatement(connection, preprocessedSelect, sql, argumentLists)
       try {
-        val resultSet = preparedStatement.executeQuery
+        val preparedStatement = prepareStatement(connection, preprocessedSelect, sql, argumentLists)
         try {
-          val result = extractor(resultSet)
-          val endTime = new DateTime
-          logger.info(s"Ran sql in ${endTime.getMillis - startTime.getMillis}ms: ${logDetails(connection, sql, argumentLists)}")
-          result
+          val resultSet = preparedStatement.executeQuery
+          try {
+            val result = extractor(resultSet)
+            val endTime = new DateTime
+            logger.info(s"Ran sql in ${endTime.getMillis - startTime.getMillis}ms: ${logDetails(connection, sql, argumentLists)}")
+            result
+          } finally {
+            try {
+              if (resultSet != null) resultSet.close
```

Comment:
I was just thinking how much we need java's try-with-resources here and looked for something like this in scala and it looks like it's just been added and will be in the 2.13 release 
_scala.util.Using_ https://github.com/scala/scala/pull/6907